BridgeBot/bidding.py

```diff
@@ -1,47 +1,53 @@
-from BridgeBot.enums import players, strains, contracts
+from enums import Players, Strains, AuctionStatus, Doubles, Contracts, PASS, Team
 
-doubles = [None, 'X', 'XX']
 
-class Auction:
-    contract = {'level': 0,
-                'strain': None,
-                'doubled': None,
-                'declarer': None}
+class InvalidDealerException(Exception):
+    pass
+
 
+class Contract:
+    def __init__(self):
+        self.level = 0
+        self.strain = Strains.NT
+        self.doubled = Doubles.NONE
+        self.declarer = None
+
+class Auction:
     def __init__(self, dealer):
         self.dealer = dealer.upper()
 
-        if not dealer in players:
-            raise Exception('Invalid dealer.')
+        if not isinstance(dealer, Players):
+            raise InvalidDealerException('Invalid dealer')
 
         self.record = {
```

Comment:
Maybe we should break Record out as its own class, since it's a data structure. It also would make it easier to deal with cases like the user accidentally inputs the wrong thing and wants an Undo.